### ISSUE TITLE
Memory-Map doesn't import GPX Files anymore (GPS Logger v2.2.10)

### ISSUE THREAD

**Issue by mikeiwalker** [BasicAirData/GPSLogger#112]
Hi, I mentioned this in some feedback on the Play Store and I like your app so much I thought I'd follow up here?
My issue is that the .gpx files now won't load into Memory Map because of the comments that've been added in this release.  I'm presuming it's an issue with MemoryMap but it was mentioned in a reply that the comments could be moved elsewhere in the file?
I've had a play around and it seems that if the comments are added in before the line:

<gpx version="1.0"

Which is basically straight after this comment:

It works just fine.  Any chance you could either move the comments there, or maybe have a configurable option to include/not include the summary in the file at all?  I certainly don't need it...

Thanks!

**Comment by GrazianoCapelli** [comment]
Hi,
thanks for opening the Issue and for giving us a Feedback of this Problem.
we are already aware of this, we mention it on #90 some Days ago thanks to some User's Feedback, and we already marked the Thing as Bug to be fixed:

> After every modification on the Exporter code, We are used to check the new KML/GPX output with ([Apache Xerces](https://xerces.apache.org/)).
> We are careful to export files 100% compatible with the GPX and KML standards.
> In fact also the Stats as Comment passed the validation Test, and then they are included into the release 2.2.10.
> 
> In spite of this, a User reports that the software "[Memory Map](https://www.memory-map.co.uk/)" doesn't accept these comments and doesn't allow to import GPX files. If the new Comment is removed, the track is imported without any problem (the GPX output of the v2.2.9 was OK).

A possible solution could be to move the stats at the Head of the file, immediately below of the existing Comments (as @mikeiwalker suggests), where they don't cause problems.

The v2.3.0 will fix this issue, please be patient some Weeks.

_In the meanwhile I suggest to report the misbehaviour also to Memory-Map's team, in order to make them aware of it too.
In fact Memory-Map doesn't open a 100% valid GPX file._

**Comment by GrazianoCapelli** [comment]
Move the Statistics at the Head of GPX file solves the Problem with Memory-Map.
The GPX head will be something like this:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!-- Created with BasicAirData GPS Logger for Android - ver. 2.3.0-dev.20201110 -->
<!-- Track 57 = 45 TrackPoints + 0 Placemarks -->

<!-- Track Statistics (based on Total Time | Time in Movement): -->
<!--  Distance = 501 m -->
<!--  Duration = 00:44 | 00:44 -->
<!--  Altitude Gap = 0 m -->
<!--  Max Speed = 54 km/h -->
<!--  Avg Speed = 41.0 | 41.0 km/h -->
<!--  Direction = ENE -->

<gpx version="1.0"
     creator="BasicAirData GPS Logger ...
     ...
```

We can implement it this Way.
The Track Statistics will be present only if the file contains a track of 1 or more Trackpoints.
The fix will be included it into the next Update v2.3.0.

**Comment by mikeiwalker** [comment]
Ace! Well done, thanks – works a treat.

 

Mike
